src/rsa.rs

```diff
@@ -289,6 +316,31 @@ impl RSAKeyPair {
     pub fn public_modulus_len(&self) -> usize {
         unsafe { RSA_size(self.rsa.as_ref()) }
     }
+
+    /// Pad the message `msg` with the padding algorithm `padding_alg` and write
+    /// the resulting signature into `signature`. The random number generattor `rng`
+    /// is used for blinding (to help make signing constant time).
+    ///
+    /// `signature` must be exactly the length of the `public_modulus_len()`.
+    pub fn sign(&self, padding_alg: &'static RSAPKCS1Padding, rng: &rand::SecureRandom,
+                msg: &[u8], signature: &mut [u8]) -> Result<(), ()> {
+        if signature.len() != self.public_modulus_len() {
+            return Err(());
+        }
+
+        let msg_len = self.public_modulus_len();
```

Comment:
I think it doesn't make sense to have this `msg_len` variable. Maybe this is an artifact of the previous iteration? If the goal is to avoid calling `public_modulus_len()` multiple times, then `msg_len` should be used in the first line of the function too. But, I think it just makes more sense to call `public_modulus_len()` below too.